pkg/kwok/controllers/template_cache.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"text/template"
+
+	"sigs.k8s.io/yaml"
+)
+
+type templateCache struct {
+	cache      sync.Map
+	bufferPool sync.Pool
+	funcMap    template.FuncMap
+}
+
+func newTemplateCache(funcMap template.FuncMap) *templateCache {
+	return &templateCache{
+		funcMap: funcMap,
+		cache:   sync.Map{},
+		bufferPool: sync.Pool{
+			New: func() interface{} {
+				return new(bytes.Buffer)
```

Comment:
```suggestion
				return bytes.NewBuffer(make([]byte, 4*1024))
```

Maybe it's better to allocate a little memory in advance